Looms.PoS/Looms.PoS.Application/Features/Reservation/Commands/CreateReservation/CreateReservationRequestValidator.cs

```diff
@@ -0,0 +1,77 @@
+using FluentValidation;
+using Looms.PoS.Application.Models.Requests.Reservation;
+using Looms.PoS.Application.Utilities.Validators;
+using Looms.PoS.Application.Utilities;
+using Looms.PoS.Domain.Interfaces;
+using Looms.PoS.Domain.Exceptions;
+
+namespace Looms.PoS.Application.Features.Reservation.Commands.CreateReservation;
+
+public class CreateReservationRequestValidator : AbstractValidator<CreateReservationRequest>
+{
+    private readonly IServicesRepository _servicesRepository;
+    private readonly IReservationsRepository _reservationsRepository;
+    public CreateReservationRequestValidator(IServicesRepository servicesRepository, IReservationsRepository reservationsRepository)
+    {
+        _servicesRepository = servicesRepository;
+        _reservationsRepository = reservationsRepository;
+
+        RuleFor(x => x.CustomerId)
+            .MustBeValidGuid()
+            .MustAsync(async (request, customerId, cancellation) =>
+                {
+                    if (Guid.TryParse(customerId, out var customerGuid))
+                    {
+                        var appointmentTime = DateTimeHelper.ConvertToUtc(request.AppointmentTime);
+                        var allReservations = await _reservationsRepository.GetAllAsync();
```

Comment:
use filter record, pass it to repository method and retrieve only reservations for specific user

Reply:
did not use filter record but maybe its fine as i pass the the vars to a repository method

Reply:
Your method checks if appointment times match, but the appointment can start sooner or later and overlap. Resolve this comment after reading since I just want to merge the changes at this point :D, but keep in mind next time